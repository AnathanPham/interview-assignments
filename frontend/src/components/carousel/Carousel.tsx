import CarouselWrap from "./CarouselWrap"

interface Props {}
const Carousel: React.FC<Props> = () => {
  return (
    <>
      <CarouselWrap>
        <h1>1</h1>
        <h1>2</h1>
        <h1>3</h1>
        <h1>4</h1>
      </CarouselWrap>
    </>
  )
}

export default Carousel
