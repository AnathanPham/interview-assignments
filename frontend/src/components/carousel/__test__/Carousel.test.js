import renderer from 'react-test-renderer';
import Carousel from '../Carousel';
import { render } from '@testing-library/react';

describe('Carousel', () => {

    beforeEach(() => {
        jest.useFakeTimers();
    });

    afterEach(() => {
        jest.useRealTimers();
    });

    it('component could be updated and unmounted without errors', () => {
        const { unmount, rerender } = render(<Carousel />);
        expect(() => {
            rerender(<Carousel />);
            unmount();
        }).not.toThrow();
    });

    it('snapshot', () => {
        const component = renderer.create(
            <Carousel></Carousel>,
        );
        let tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    })
})

