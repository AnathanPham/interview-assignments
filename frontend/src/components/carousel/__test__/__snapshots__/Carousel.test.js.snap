// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Carousel snapshot 1`] = `
<div
  className="carousel-wrap"
>
  <div
    className="carousel-track"
    style={
      Object {
        "transform": "translate3d(-000%, 0, 0)",
      }
    }
  >
    <div
      className="carousel-slide"
      data-index={0}
      style={Object {}}
      tabIndex="-1"
    >
      <div
        className="carousel-banner"
        style={
          Object {
            "color": "#fff",
          }
        }
      >
        <div
          className="carousel-banner__placeholder-top"
        />
        <h1
          className="carousel-banner__title"
        >
          xPhone
        </h1>
        <h2
          className="carousel-banner__subtitle"
        >
          Lots to love.Less to spend.
        </h2>
        <h2
          className="carousel-banner__subtitle"
        >
          Starting at $399
        </h2>
        <img
          alt="xPhone"
          className="carousel-banner__image"
          src="iphone.png"
        />
        <div
          className="carousel-banner__background"
          style={
            Object {
              "backgroundColor": "#111",
            }
          }
        />
      </div>
    </div>
    <div
      className="carousel-slide"
      data-index={1}
      style={Object {}}
      tabIndex="-1"
    >
      <div
        className="carousel-banner"
        style={
          Object {
            "color": "#000",
          }
        }
      >
        <div
          className="carousel-banner__placeholder-top"
        />
        <h1
          className="carousel-banner__title"
        >
          Tablet
        </h1>
        <h2
          className="carousel-banner__subtitle"
        >
          Just the right amount of everything.
        </h2>
        <img
          alt="Tablet"
          className="carousel-banner__image"
          src="tablet.png"
        />
        <div
          className="carousel-banner__background"
          style={
            Object {
              "backgroundColor": "#fafafa",
            }
          }
        />
      </div>
    </div>
    <div
      className="carousel-slide"
      data-index={2}
      style={Object {}}
      tabIndex="-1"
    >
      <div
        className="carousel-banner"
        style={
          Object {
            "color": "#000",
          }
        }
      >
        <div
          className="carousel-banner__placeholder-top"
        />
        <h1
          className="carousel-banner__title"
        >
          Buy a Tablet or xPhone for college.
        </h1>
        <h1
          className="carousel-banner__title"
        >
          get arPods.
        </h1>
        <img
          alt="Buy a Tablet or xPhone for college.get arPods."
          className="carousel-banner__image"
          src="airpods.png"
        />
        <div
          className="carousel-banner__background"
          style={
            Object {
              "backgroundColor": "#f1f1f3",
            }
          }
        />
      </div>
    </div>
  </div>
  <div
    className="carousel-dots"
  >
    <div
      className="carousel-dot"
    >
      <div
        className="carousel-dot__process"
        style={Object {}}
      />
    </div>
    <div
      className="carousel-dot"
    >
      <div
        className="carousel-dot__process"
        style={Object {}}
      />
    </div>
    <div
      className="carousel-dot"
    >
      <div
        className="carousel-dot__process"
        style={Object {}}
      />
    </div>
  </div>
</div>
`;

exports[`can render 1`] = `
<div
  className="carousel-wrap"
>
  <div
    className="carousel-track"
    style={
      Object {
        "transform": "translate3d(-000%, 0, 0)",
      }
    }
  >
    <div
      className="carousel-slide"
      data-index={0}
      style={Object {}}
      tabIndex="-1"
    >
      <div
        className="carousel-banner"
        style={
          Object {
            "color": "#fff",
          }
        }
      >
        <div
          className="carousel-banner__placeholder-top"
        />
        <h1
          className="carousel-banner__title"
        >
          xPhone
        </h1>
        <h2
          className="carousel-banner__subtitle"
        >
          Lots to love.Less to spend.
        </h2>
        <h2
          className="carousel-banner__subtitle"
        >
          Starting at $399
        </h2>
        <img
          alt="xPhone"
          className="carousel-banner__image"
          src="iphone.png"
        />
        <div
          className="carousel-banner__background"
          style={
            Object {
              "backgroundColor": "#111",
            }
          }
        />
      </div>
    </div>
    <div
      className="carousel-slide"
      data-index={1}
      style={Object {}}
      tabIndex="-1"
    >
      <div
        className="carousel-banner"
        style={
          Object {
            "color": "#000",
          }
        }
      >
        <div
          className="carousel-banner__placeholder-top"
        />
        <h1
          className="carousel-banner__title"
        >
          Tablet
        </h1>
        <h2
          className="carousel-banner__subtitle"
        >
          Just the right amount of everything.
        </h2>
        <img
          alt="Tablet"
          className="carousel-banner__image"
          src="tablet.png"
        />
        <div
          className="carousel-banner__background"
          style={
            Object {
              "backgroundColor": "#fafafa",
            }
          }
        />
      </div>
    </div>
    <div
      className="carousel-slide"
      data-index={2}
      style={Object {}}
      tabIndex="-1"
    >
      <div
        className="carousel-banner"
        style={
          Object {
            "color": "#000",
          }
        }
      >
        <div
          className="carousel-banner__placeholder-top"
        />
        <h1
          className="carousel-banner__title"
        >
          Buy a Tablet or xPhone for college.
        </h1>
        <h1
          className="carousel-banner__title"
        >
          get arPods.
        </h1>
        <img
          alt="Buy a Tablet or xPhone for college.get arPods."
          className="carousel-banner__image"
          src="airpods.png"
        />
        <div
          className="carousel-banner__background"
          style={
            Object {
              "backgroundColor": "#f1f1f3",
            }
          }
        />
      </div>
    </div>
  </div>
  <div
    className="carousel-dots"
  >
    <div
      className="carousel-dot"
    >
      <div
        className="carousel-dot__process"
        style={Object {}}
      />
    </div>
    <div
      className="carousel-dot"
    >
      <div
        className="carousel-dot__process"
        style={Object {}}
      />
    </div>
    <div
      className="carousel-dot"
    >
      <div
        className="carousel-dot__process"
        style={Object {}}
      />
    </div>
  </div>
</div>
`;

exports[`snapshot test 1`] = `
<div
  className="carousel-wrap"
>
  <div
    className="carousel-track"
    style={
      Object {
        "transform": "translate3d(-000%, 0, 0)",
      }
    }
  >
    <div
      className="carousel-slide"
      data-index={0}
      style={Object {}}
      tabIndex="-1"
    >
      <div
        className="carousel-banner"
        style={
          Object {
            "color": "#fff",
          }
        }
      >
        <div
          className="carousel-banner__placeholder-top"
        />
        <h1
          className="carousel-banner__title"
        >
          xPhone
        </h1>
        <h2
          className="carousel-banner__subtitle"
        >
          Lots to love.Less to spend.
        </h2>
        <h2
          className="carousel-banner__subtitle"
        >
          Starting at $399
        </h2>
        <img
          alt="xPhone"
          className="carousel-banner__image"
          src="iphone.png"
        />
        <div
          className="carousel-banner__background"
          style={
            Object {
              "backgroundColor": "#111",
            }
          }
        />
      </div>
    </div>
    <div
      className="carousel-slide"
      data-index={1}
      style={Object {}}
      tabIndex="-1"
    >
      <div
        className="carousel-banner"
        style={
          Object {
            "color": "#000",
          }
        }
      >
        <div
          className="carousel-banner__placeholder-top"
        />
        <h1
          className="carousel-banner__title"
        >
          Tablet
        </h1>
        <h2
          className="carousel-banner__subtitle"
        >
          Just the right amount of everything.
        </h2>
        <img
          alt="Tablet"
          className="carousel-banner__image"
          src="tablet.png"
        />
        <div
          className="carousel-banner__background"
          style={
            Object {
              "backgroundColor": "#fafafa",
            }
          }
        />
      </div>
    </div>
    <div
      className="carousel-slide"
      data-index={2}
      style={Object {}}
      tabIndex="-1"
    >
      <div
        className="carousel-banner"
        style={
          Object {
            "color": "#000",
          }
        }
      >
        <div
          className="carousel-banner__placeholder-top"
        />
        <h1
          className="carousel-banner__title"
        >
          Buy a Tablet or xPhone for college.
        </h1>
        <h1
          className="carousel-banner__title"
        >
          get arPods.
        </h1>
        <img
          alt="Buy a Tablet or xPhone for college.get arPods."
          className="carousel-banner__image"
          src="airpods.png"
        />
        <div
          className="carousel-banner__background"
          style={
            Object {
              "backgroundColor": "#f1f1f3",
            }
          }
        />
      </div>
    </div>
  </div>
  <div
    className="carousel-dots"
  >
    <div
      className="carousel-dot"
    >
      <div
        className="carousel-dot__process"
        style={Object {}}
      />
    </div>
    <div
      className="carousel-dot"
    >
      <div
        className="carousel-dot__process"
        style={Object {}}
      />
    </div>
    <div
      className="carousel-dot"
    >
      <div
        className="carousel-dot__process"
        style={Object {}}
      />
    </div>
  </div>
</div>
`;
